server/socket-manager.js

```diff
@@ -10,11 +14,33 @@ const authenticateCookie = (socket, callback) => {
 };
 
 const socketManager = (socket) => {
-  socket.on('authenticate', () => {
-    authenticateCookie(socket, (err, data) => {
-      if(err) socket.emit('User not authenticated');
-      socket.emit('userdata', data);
-      // handles socket events
+  let user_id = '';
+
+  socket.on('authenticate', (_, clientCallback) => {
```

Comment:
@des-des not sure where the problem is... it works (so far)! here're the docs I found:
https://socket.io/docs/#sending-and-getting-data-(acknowledgements)